scripts/inference/ft/mpt_ckpt_to_ft.py

```diff
@@ -0,0 +1,294 @@
+        'norm_1.weight': 'input_layernorm.weight',
+        'attn.Wqkv.bias': 'attention.query_key_value.bias',
+        'attn.Wqkv.weight': 'attention.query_key_value.weight',
+        'attn.out_proj.bias': 'attention.dense.bias',
+        'attn.out_proj.weight': 'attention.dense.weight',
+        'norm_2.bias': 'post_attention_layernorm.bias',
+        'norm_2.weight': 'post_attention_layernorm.weight',
+        'ffn.up_proj.bias': 'mlp.dense_h_to_4h.bias',
+        'ffn.up_proj.weight': 'mlp.dense_h_to_4h.weight',
+        'ffn.down_proj.bias': 'mlp.dense_4h_to_h.bias',
+        'ffn.down_proj.weight': 'mlp.dense_4h_to_h.weight',
+    }
+
+    pool = multiprocessing.Pool(args.processes)
+    for name, param in model.named_parameters():
+        print(f'Workign on parameter {name} ...')
+        data = param.detach().cpu().numpy().astype(np_weight_data_type)
+        if name.find('weight') == -1 and name.find('bias') == -1:
+            print(f'found a parameter name that is not handled: {name}')
+            continue
+        if name == 'transformer.wpe.weight':
+            assert data.shape == (
+                hf_config['max_seq_len'],
+                hf_config['d_model']), f'unexpected dim for {name}'
+            data.tofile(save_dir + 'model.wpe.bin')
+        elif name == 'transformer.wte.weight':
+            assert data.shape == (
+                hf_config['vocab_size'],
+                hf_config['d_model']), f'unexpected dim for {name}'
+            data.tofile(save_dir + 'model.wte.bin')
+        elif name == 'transformer.norm_f.bias':
+            assert data.shape == (
+                hf_config['d_model'],), f'unexpected dim for {name}'
+            data.tofile(save_dir + 'model.final_layernorm.bias.bin')
+        elif name == 'transformer.norm_f.weight':
+            assert data.shape == (
+                hf_config['d_model'],), f'unexpected dim for {name}'
+            save_path = save_dir + 'model.final_layernorm.weight.bin'
+            data.tofile(save_path)
+            if hf_config['no_bias']:
+                write_empty_bias(name, save_path, data.shape[-1])
+        elif name == 'transformer.lm_head.weight':
+            data.tofile(save_dir + 'model.lm_head.weight.bin')
+        else:
+            for mpt_pattern, ft_pattern in param_remapping.items():
+                if name.find(mpt_pattern) != -1:
+                    new_name = name.replace('transformer.blocks.',
+                                            'layers.').replace(
+                                                mpt_pattern, ft_pattern)
+                    pool.starmap(
+                        split_and_convert_process,
+                        [(0, save_dir, factor, new_name, hf_config, data)],
+                    )
+
+    pool.close()
+    pool.join()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-save_dir',
```

Comment:
I think the long argument names usually have `--`

Reply:
changed.